src/db/mongo/MongoInterface.cpp

```diff
@@ -81,10 +81,15 @@ MongoInterface::MongoInterface()
 	bson_error_t err;
 	std::string mongo_uri;
 	if(!rosprolog_kb::node().getParam(std::string("mongodb_uri"),mongo_uri)) {
-		char *mongo_host = std::getenv("MONGO_PORT_27017_TCP_ADDR");
+		char *mongo_host = std::getenv("KNOWROB_MONGO_HOST");
 		if(mongo_host != NULL) {
+			char *mongo_user_name = std::getenv("KNOWROB_MONGO_USER");
+			char *mongo_user_pw = std::getenv("KNOWROB_MONGO_PASS");
+			char *mongo_database = std::getenv("KNOWROB_MONGO_DB");
```

Comment:
but there is a setting to configure this *mng_client:db_name*, can't we use this instead of env?

Reply:
why to append the DB name anyway? I think this is not needed

Reply:
maybe it would be good to provide an interface to read settings in C++ code, basically doing a call of `setting/2` in the Pl engine

Reply:
> but there is a setting to configure this _mng_client:db_name_, can't we use this instead of env?
> maybe it would be good to provide an interface to read settings in C++ code, basically doing a call of setting/2 in the Pl engine

I haven't found an easy way to access the pl settings. I thought that to get them from the pl engine i had to start a PL engine in the already running PL Engine.

> why to append the DB name anyway? I think this is not needed
 
The db name is important for the authentication. If no db is in the string, the user authenticates agains the admin database, which will not work in our case

Reply:
no it should be possible straight forward, I think.
Maybe there is something strange with using C++ API of swipl, I had a strange situation where I thought that the C++ API does not like to do queries using the Pl engines managed by rosprolog. However, I had no issue with using the C API.

Reply:
This is an example for using the C++ API:

```C++
  PlQuery q("call", PlTermv(A2));
  while( q.next_solution() )
  { sum += (long)A1;
    n++;
  }
```

Reply:
here is an example for C API: https://github.com/daniel86/rosprolog/blob/master/src/rosprolog_node/PrologNode.cpp#L61